src/parameterizations/lateral/MOM_MEKE.F90

```diff
@@ -557,12 +561,15 @@ subroutine MEKE_equilibrium(CS, MEKE, G, GV, US, SN_u, SN_v, drag_rate_visc, I_m
   real :: beta, SN, bottomFac2, barotrFac2, LmixScale, Lrhines, Leady
   real :: I_H, KhCoeff, Kh, Ubg2, cd2, drag_rate, ldamping, src
   real :: EKE, EKEmin, EKEmax, resid, ResMin, ResMax, EKEerr
+  real, dimension(SZI_(G),SZJ_(G)) :: D, lat, lon !< ocean depth, lat, & lon at h-points
+  real :: FatH    ! Coriolis parameter at h points; to compute topographic beta
   integer :: i, j, is, ie, js, je, n1, n2
   real, parameter :: tolerance = 1.e-12 ! Width of EKE bracket in m^2 s^-2.
   logical :: useSecant, debugIteration
 
   is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec
-
+  D = G%bathyT; lat = G%geolatt; lon = G%geolont
```

Comment:
I don't think you need to, or should, copy these arrays here:
- `lon` and `lat` aren't used
- Copying `D` is slow and if you don't modify it then is there even a need to copy it at all? You could just reference `G%bathyT(...)` where you have `D(...)`
- Related (but redundant): we prefer `D(:,:) = G%bathyT(:,:)`, or loops, to the scalar notation. This makes it is clear the data is an array.